knowledge/pkg/datastore/documentloader/converter/soffice.go

```diff
@@ -53,20 +53,35 @@ func (c *SofficeConverter) Convert(ctx context.Context, reader io.Reader, source
 	}
 	_ = tempfile.Close()
 
+	logger := log.FromCtx(ctx)
+
+	profileDir, err := os.MkdirTemp(os.TempDir(), "libreoffice-profile-*")
```

Comment:
Does it make sense to clean up any of these directories/files?

Also, it looks like this is just an empty file? Is that intended?

Reply:
Cleanup is deferred: https://github.com/otto8-ai/tools/pull/249/files/4bd30a02c44e367c1321019c483c41f8f5ebe993#diff-a3f56052448642b16aaeabd7d8a58f4e7c4251780db6ce377f91dbd915d48eaaR63

> Also, it looks like this is just an empty file? Is that intended?

It's just an empty dir, that's correct. `soffice` initializes it on start.
In fact, the dir doesn't even need to exist before - it just must not be used by another instance of libreoffice at the same time.
There could be smarter solutions like using a server or just some "connection pooling" with a fixed set of profiles, but I'd like to defer that extra engineering overhead for now.